modules/apps/transfer/ibc_module.go

```diff
@@ -301,3 +306,15 @@ func (im IBCModule) OnTimeoutPacket(
 
 	return nil
 }
+
+// UnmarshalPacketData attempts to unmarshal the provided packet data bytes
+// into a FungibleTokenPacketData. This function implements the optional
+// PacketDataUnmarshaler interface required for ADR 008 support.
+func (im IBCModule) UnmarshalPacketData(bz []byte) (interface{}, error) {
```

Comment:
We could now also re-use this function [here](https://github.com/cosmos/ibc-go/blob/abdc496945c666da866f1ab24142210daab0b816/modules/apps/transfer/ibc_module.go#L177-L183) and in a few other places. Same for the function added in ICA. Can be addressed in a separate PR. :)